app/spandx.js

```diff
@@ -77,7 +77,13 @@ async function init(confIn) {
     }
 
     if (_.get(conf, "esi")) {
-        app.use(transformerProxy(createEsiMiddleware(conf)));
+        app.use((res, req, next) => {
+            if (res.headers.accept === 'text/event-stream') {
```

Comment:
Instead of targeting `text/event-stream` specifically, I'd like for this if to be flipped around.  Only run `transformerProxy(createEsiMiddleware(...))` when `res.headers.accept === "text/html"`.  ESI shouldn't run on anything else.  I think that will fix your issue and will likely fix any adjacent, undiscovered issues like yours.

Reply:
Yeah, you are right! Totally doable. I've checked what browsers add by default to general requests and looks like they don't include `text/html` but rather some default values `image/webp,image/apng,image/*,*/*;q=0.8` I guess the safest way here is to use esiMiddleware when `*/*` appears in the accept header or when `text/html` is there. What do you think?

Reply:
Hmm, tough call.  What you found makes total sense.  If you open a new tab and paste in a URL, it could be any type of resource, so the broad Accept header is understandable.

Let's go with your suggestion, I think it sounds solid.  We can refine it later if any issues emerge.

Reply:
Awesome! Updated.